Library/Homebrew/os/linux.rb

```diff
@@ -1,6 +1,7 @@
 # typed: true
 # frozen_string_literal: true
 
+require "os/mac/version"
```

Comment:
I think we should avoid `os/mac` includes in this file.

Reply:
Then where do we get `OS::Mac::Version` from?

Reply:
@reitermarkus How has it been working before now? Ideally we wouldn't even define `module Mac` below at all but I feel pretty strongly about not introducing more cross-OS coupling like this.

Reply:
Wrote up https://github.com/Homebrew/brew/issues/15343

Reply:
Before it was using just `Version::NULL` instead of `OS::Mac::Version::NULL` and some macOS-specific methods were implemented on `NullVersion`.

Reply:
I think having a single `Version::NULL` would be better here, particularly if it avoids this include.

Reply:
`Version::NULL` doesn't have the macOS-specific methods.

Reply:
Something was working before this PR without this `require`. I'm not going to ✅ this PR with this `require`, sorry, given https://github.com/Homebrew/brew/issues/15343. It's a step in the wrong direction.

Reply:
Well, before this, Linux was returning a different type here. Returning the same type on macOS and Linux is objectively better for type-checking than having basically `T.any(OS::Mac::Version, NullVersion)` here.

Even if https://github.com/Homebrew/brew/issues/15343 makes sense, `OS::Mac::full_version` and `OS::Mac::version` in `os/linux` can simply be removed *after* deprecating them. That doesn't have to block this PR.

Reply:
> Well, before this, Linux was returning a different type here. Returning the same type on macOS and Linux is objectively better for type-checking than having basically `T.any(OS::Mac::Version, NullVersion)` here.

That's fine with me if it avoids this `require`.

Reply:
That doesn't work because `Version::NULL` doesn't have the macOS-specific methods because `NullVersion` is integrated into `Version` and it doesn't make sense for every `Version` to have macOS-specific methods.

Reply:
> it doesn't make sense for every Version to have macOS-specific methods

This makes more sense to me than this require.

Reply:
Not sure what else to say here other than I disagree.

So this PR is blocked until https://github.com/Homebrew/brew/issues/15343 is closed then?

Reply:
Note that `OS::Mac::Version` is already `require`d on Linux anyways since `Utils::Bottles::Tag#to_macos_version` and `SoftwareSpec#uses_from_macos` use it, so adding this `require` doesn't change anything.

Reply:
> So this PR is blocked until https://github.com/Homebrew/brew/issues/15343 is closed then?

If you're not willing to make this change: yes.

Note: that issue is not going to get closed so long as I remain project leader of Homebrew unless the TSC overrules me here as per: https://docs.brew.sh/Homebrew-Leadership-Responsibilities#pl-sole-responsibilities

Reply:
I am sorry: again, I am not willing to ✅ this PR with this `require` in place. It's up to you to decide what to do with/about that but I'm not sure continued discussion here is going to achieve anything there.

Reply:
I just don't get your reasoning here. What is so special about `require "os/mac/version"`? Is it only the path that is bothering you?

The implementation of `OS::Mac::Version` does not actually depend on macOS. Are you fine with renaming it to `MacOSVersion` and moving it to `macos_version.rb` and then doing `require "macos_version"`?

Reply:
> I just don't get your reasoning here. What is so special about require "os/mac/version"? Is it only the path that is bothering you?

I don't want to have any more requires in `os/linux` to stuff in `os/mac`.

> Are you fine with renaming it to MacOSVersion and moving it to macos_version.rb and then doing require "macos_version"?

Please just leave this methods on `Version` like they were before.

If you're unwilling to do that: I'd rather not have a `MacOSVersion` but it's definitely preferable to doing a `require` from `os/mac`, yes.

Reply:
> Please just leave this methods on `Version` like they were before.

They weren't in `Version` before, but in `NullVersion`. In this PR, those methods are merged into `Version` and `OS::Mac::Version`, respectively. By moving the macOS-specific methods into `Version`, we would duplicate them again, defeating the main point of this PR.

I will go ahead and move it out of `os/mac`.